cli/initiator/reshare.go

```diff
@@ -15,9 +17,98 @@ import (
 )
 
 func init() {
+	cli_utils.SetGenerateReshareMsgFlags(GenerateReshareMsg)
 	cli_utils.SetReshareFlags(StartReshare)
 }
 
+var GenerateReshareMsg = &cobra.Command{
+	Use:   "generate-reshare-msg",
+	Short: "Generate reshare message for one or multiple ceremonies",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := cli_utils.SetViperConfig(cmd); err != nil {
+			return err
+		}
+		if err := cli_utils.BindGenerateReshareMsgFlags(cmd); err != nil {
+			return err
+		}
+		logger, err := cli_utils.SetGlobalLogger(cmd, "dkg-initiator")
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := cli_utils.Sync(logger); err != nil {
+				log.Printf("Failed to sync logger: %v", err)
+			}
+		}()
+		logger.Info("🪛 Initiator`s", zap.String("Version", cmd.Version))
+		opMap, err := cli_utils.LoadOperators(logger)
+		if err != nil {
+			logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
+		}
+		oldOperatorIDs, err := cli_utils.StringSliceToUintArray(cli_utils.OperatorIDs)
+		if err != nil {
+			logger.Fatal("😥 Failed to load participants: ", zap.Error(err))
+		}
+		newOperatorIDs, err := cli_utils.StringSliceToUintArray(cli_utils.NewOperatorIDs)
+		if err != nil {
+			logger.Fatal("😥 Failed to load new participants: ", zap.Error(err))
+		}
+		// create initiator instance
+		dkgInitiator, err := initiator.New(opMap.Clone(), logger, cmd.Version, cli_utils.ClientCACertPath)
+		if err != nil {
+			return err
+		}
+		var signedProofs [][]*spec.SignedProof
+		if cli_utils.ProofsFilePath != "" {
+			signedProofs, err = wire.LoadProofs(cli_utils.ProofsFilePath)
+			if err != nil {
+				logger.Fatal("😥 Failed to read proofs json file:", zap.Error(err))
+			}
+		}
+		if cli_utils.ProofsString != "" {
+			signedProofs, err = cli_utils.DecodeProofsString(cli_utils.ProofsString)
+			if err != nil {
+				logger.Fatal("😥 Failed to read proofs string:", zap.Error(err))
+			}
+		}
+		ethNetwork := e2m_core.NetworkFromString(cli_utils.Network)
+		if ethNetwork == "" {
+			logger.Fatal("😥 Cant recognize eth network")
+		}
+		rMsgs := []*wire.ReshareMessage{}
+		for i := 0; i < len(signedProofs); i++ {
+			nonce := cli_utils.Nonce + uint64(i)
+			// Contruct the resign message
+			rMsg, err := dkgInitiator.ConstructReshareMessage(
+				oldOperatorIDs,
+				newOperatorIDs,
+				signedProofs[i][0].Proof.ValidatorPubKey,
+				ethNetwork,
+				cli_utils.WithdrawAddress[:],
+				cli_utils.OwnerAddress,
+				nonce,
+				signedProofs[i],
+			)
```

Comment:
do I understand correctly that bulk reshare always maps from a single set of operators to another? 

I just want to make sure this is what product intended @MatusKysel @RaekwonIII?
It probably is but just making sure